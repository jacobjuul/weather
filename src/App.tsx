import React from 'react';

import { Weather } from './Weather';

import type { FC } from 'react';

import './app.css';

export const App: FC = () => {
  const [input, setInput] = React.useState('');
  if (input) {
     const [city, setCity] = React.useState('');
  }

  const doStuff = (event: React.ChangeEvent<HTMLInputElement>) => {
    setInput(event.target.value);
  };


  return (
    <>
      <input
        role="search"
        type="text"
        value={input}
        onChange={doStuff}
      />

      <button onClick={() => setCity(input)}>Show Weather</button>

      <Weather city={city} />
    </>
  );
};
