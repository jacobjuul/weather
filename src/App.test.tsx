import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { App } from './App';

describe('App', () => {
    it('Renders Component According To Snapshot', () => {
        const { container } = render(<App />);

        expect(container).toMatchSnapshot();
    });

    it('Checks Input Functionality', async () => {
        const user = userEvent.setup();
        render(<App />);

        const searchInput = screen.getByRole('search');
        await user.type(searchInput, 'Copenhagen');
        expect(searchInput).toHaveValue('Copenhagen');
    });
});
