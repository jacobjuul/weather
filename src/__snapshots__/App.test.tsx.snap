// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > Renders Component According To Snapshot 1`] = `
<div>
  <input
    role="search"
    type="text"
    value=""
  />
  <button>
    Show Weather
  </button>
  <h1 />
  <p>
     Celcius
  </p>
  <img
    alt="Icon"
  />
</div>
`;
